roles/kubespray-defaults/defaults/main/main.yml

```diff
@@ -437,15 +437,15 @@ argocd_enabled: false
 
 ## When OpenStack is used, Cinder version can be explicitly specified if autodetection fails (Fixed in 1.9: https://github.com/kubernetes/kubernetes/issues/50461)
 # openstack_blockstorage_version: "v1/v2/auto (default)"
-openstack_blockstorage_ignore_volume_az: "{{ volume_cross_zone_attachment | default('false') }}"
+openstack_blockstorage_ignore_volume_az: "{{ volume_cross_zone_attachment }}"
```

Comment:
when this role is executed, nothing say that volume_cross_zone_attachment is defined. 
this variable eventually is defined but from another role/inventory. 
We shouldn't remove it there

Reply:
Default variables are lazily interpreted though, aren't they ? (-> meaning this will work if the value is defined when the variable is used)

Reply:
Nothing tells you it will be defined for sure. 
Maybe it's defined in the openstack role that is not included when not using openstack, 
Maybe the playbook ran with --tags  X,Y and the role where "volume_cross_zone_attachment" is not executed.

Reply:
<del>

We're mostly doing role imports rather than include, though, don't we ?

If I'm not mistaken, the tags don't affect roles variables, only the tasks (unless using include, which is dynamic).

However, in any case, I agree it's better than a variable dependent on another should live nearby, ideally in the same file. ~~

</del>

Actually, this variable is not used anywhere since 437026f (AFAICS), so we should just delete it :
```ShellSession
$ grep -R openstack_blockstorage_ignore roles/
roles/kubespray-defaults/defaults/main/main.yml:openstack_blockstorage_ignore_volume_az: "{{ volume_cross_zone_attachment | default('false') }}"
```

Reply:
@VannTen yes, but not all runs imports all roles.

Short test:

role A define "var_a: "I'm var a"
role B define "var_ab: "I'm b and {{var_a}}"

In the test playbook, the `install kubernetes nodes` run will work , but not the `install control plane`
```
---
- name: Install Kubernetes nodes
  hosts: localhost
  gather_facts: true
  roles:
    - { role: a, tags: ['a'], when: false}
    - { role: b, tags: ['b'] }


- name: Install the control plane
  hosts: localhost
  gather_facts: true
  roles:
    - { role: b, tags: ['b'] }
```
Selecting subset of roles is used a lot in kubespray (see cluster.yml) 
![image](https://github.com/user-attachments/assets/3d81a0c5-2aa5-4867-871e-65ed6b832235)